core/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -346,6 +346,86 @@ class ApplicationInfo(
     }
   }
 
+  /**
+   * This function is meant to clean up Delta log execs so that you could align
+   * SQL ids between CPU and GPU eventlogs. It attempts to remove any delta log
+   * only SQL ids. This includes reading checkpoints, delta_log json files,
+   * updating Delta state cache/table. There is also one special case for
+   * MergeIntoCommandEdge that I think showed up because some eventlogs might not
+   * be on the exact same data.
+   */
+  def cleanSQLIdsToAlign(): Seq[SQLCleanAndAlignIdsProfileResult] = {
+    val noDeltaLogs = sqlPlans.filter {
+      case (id, planInfo) =>
+        val planGraph = ToolsPlanGraph(planInfo)
+        val sqlDesc = sqlIdToInfo(id).physicalPlanDesc
+        val leftoverNodes = planGraph.allNodes.filter { node =>
+          val normalizedNodeName = node.name.stripSuffix("$")
+          normalizedNodeName match {
+            case "RDDScanExec" =>
+              if (sqlDesc.contains("Delta Table State") ||
+                sqlDesc.contains("Delta Table Checkpoint") ||
+                sqlDesc.contains("delta_log")) {
+                false
+              } else {
+                true
+              }
+            case s if s.contains("WholeStageCodegen") =>
+              if (sqlDesc.contains("_delta_log")) {
+                false
+              } else {
+                true
+              }
+            case s if s.contains("MergeIntoCommandEdge") =>
+              // this is a bit odd but GPU doesn't accelerate anyway, I think this might be
+              // due to differences in data between runs
+              // Execute MergeIntoCommandEdge (1)
+              //   +- MergeIntoCommandEdge (2)
+              if (planGraph.allNodes.size < 2) {
+                false
+              } else {
+                true
+              }
+            case "LocalTableScan" =>
+              if (sqlDesc.contains("stats_parsed.numRecords")) {
+                false
+              } else if (planGraph.allNodes.size == 1) {
+                false
+              } else {
+                true
+              }
+            case s if ReadParser.isScanNode(s) =>
+              if (sqlDesc.contains("_delta_log") || sqlDesc.contains("_databricks_internal")) {
+                false
+              } else if (sqlDesc.contains("checkpoint")) {
+                // double check it has parquet - regex are expensive though so only do
+                // if necessary
+                val checkpoint = ".*Location:(.*)checkpoint(.*).parquet(.*)".r
+                if (checkpoint.findFirstIn(sqlDesc).isDefined) {
+                  false
+                } else {
+                  true
+                }
+              } else {
+                true
+              }
+            case _ =>
+              true
+          }
+        }
+        if (leftoverNodes.size < planGraph.allNodes.size) {
+          false
+        } else {
+          true
+        }
+      case _ =>
```

Comment:
Should put another to act as a space holder for other Databricks ops for future expansion?
We can have a Map[Op, Boolean] to mark which ones True/False.
This map can be updated whenever we discover a corner case, or when the RAPIDS plugin adds a new support to a delta operator:
Example, `UpdateCommandEdge`, `DeleteCommandEdge`, ..etc.